static/app/types/auth.tsx

```diff
@@ -100,6 +102,11 @@ export type ChallengeData = {
   webAuthnRegisterData: string;
 };
 
+export type AuthStateData = {
+  challenge: string;
+  user_verification: 'required' | 'preferred' | 'discouraged' | null;
```

Comment:
I'd recommend pulling the `required` etc into an enum, then making `user_verification` optional so you don't need the `| null`.

```ts
// is there a better name for this? AuthStateVerification?
export enum UserVerificationState = {
   REQUIERD: 'required',
   PREFERRED: 'preferred',
   DISCOURAGED: 'discouraged',
};
```

```suggestion
  user_verification?: UserVerificationState;
```
This will give a bit better type error information, and be a bit closer to the represented state (rather than null, we don't have it yet)